ballerina/tests/mock_service.bal

```diff
@@ -0,0 +1,288 @@
+        // Mock response with an array of Filter objects
+      return [
+            {
+                "id": "10000",
+                "name": "My Favorite Filter",
+                "description": "This is a mock description for the favorite filter."
+            },
+            {
+                "id": "10001",
+                "name": "Another Favorite Filter",
+                "description": "Description for another favorite filter."
+            }
+        ];
+    }
+
+    resource function get rest/api/'3/filter/my(string? expand, boolean includeFavourites = false) returns Filter[]|http:Unauthorized {
+        // Mock response with an array of Filter objects
+       return [
+            {
+                "id": "20000",
+                "name": "My Mock Filter",
+                "description": "This is a mock description for my filter."
+            },
+            {
+                "id": "20001",
+                "name": "Another Mock Filter",
+                "description": "Description for another mock filter."
+            }
+        ];
+    }
+
+    resource function post rest/api/'3/group(@http:Payload AddGroupBean payload) returns Group|http:BadRequest|http:Unauthorized|http:Forbidden {
+            return {
+                "name": "first group",
+                "groupId": "12345",
+                "self": "http://localhost:9090/rest/api/3/group?groupId=12345"
+            };
+    }
+
+    resource function delete rest/api/'3/group(string? groupname, string? groupId, string? swapGroup, string? swapGroupId) returns http:Ok|http:BadRequest|http:Unauthorized|http:Forbidden|http:NotFound {
+        http:Response mockResponse = new;
+        mockResponse.statusCode = 200;
+        mockResponse.setHeader("x-mock-confirmation", "Group 'first group' deleted successfully");
+        return mockResponse;
+    }
+
+    resource function get rest/api/'3/fieldconfiguration(int[]? id, int startAt = 0, int:Signed32 maxResults = 50, boolean isDefault = false, string query = "") returns PageBeanFieldConfigurationDetails|http:Unauthorized|http:Forbidden {
+        // Create a mock response
+        json mockResponse = {
+            "startAt": 0,
+            "maxResults": 50,
+            "total": 10,  // Total field configurations, set to >0 for testing
+            "values": [
+                {
+                    "id": "1",
+                    "name": "Default Field Configuration",
+                    "description": "The default field configuration",
+                    "isDefault": true
+                },
+                {
+                    "id": "2",
+                    "name": "Custom Field Configuration",
+                    "description": "A custom field configuration",
+                    "isDefault": false
+                }
+            ]
+        };
+
+        // Send the mock response
+        http:Response res = new;
+        res.statusCode = 200;
+        res.setPayload(mockResponse);
+        return(res);
```

Comment:
This function also should give a compilation error.

Reply:
And as far as I saw this json response does not align with the `PageBeanFieldConfigurationDetails` response. Please check that as well

Reply:
Please make sure to build the module and run the tests locally